Swiftification/Array.swift

```diff
@@ -99,26 +99,26 @@ public extension Array {
     }
     
     /// Sections the ordered array by the string specified in the closure, preserving the order of the original array.
-    func sectionBy(@noescape sectionFunction: (Element) -> String) -> [(title: String, items: [Element])] {
-        var lastTitle = ""
+    func sectionBy<T: Equatable>(@noescape sectionFunction: (Element) -> T) -> [(T, [Element])] {
+        var lastTitle: T? = nil
```

Comment:
`Name` implies `String`, but `Header` doesn't, so I like the `Header` variants better.